src/util/simplify_expr_int.cpp

```diff
@@ -1101,34 +1103,32 @@ bool simplify_exprt::simplify_power(exprt &expr)
 }
 
 /// Simplifies extracting of bits from a constant.
-bool simplify_exprt::simplify_extractbits(exprt &expr)
+bool simplify_exprt::simplify_extractbits(extractbits_exprt &expr)
 {
-  assert(expr.operands().size()==3);
-
-  const typet &op0_type=expr.op0().type();
+  const typet &op0_type = expr.src().type();
 
   if(!is_bitvector_type(op0_type) &&
      !is_bitvector_type(expr.type()))
     return true;
 
-  if(expr.op0().is_constant())
-  {
-    std::size_t width=to_bitvector_type(op0_type).get_width();
-    mp_integer start, end;
+  mp_integer start, end;
 
-    if(to_integer(expr.op1(), start))
-      return true;
+  if(to_integer(expr.upper(), start))
+    return true;
 
-    if(to_integer(expr.op2(), end))
-      return true;
+  if(to_integer(expr.lower(), end))
+    return true;
 
-    if(start<0 || start>=width ||
-       end<0 || end>=width)
-      return true;
+  const std::size_t width = to_bitvector_type(op0_type).get_width();
+
+  if(start < 0 || start >= width || end < 0 || end >= width)
+    return true;
 
-    assert(start>=end); // is this always the case??
+  PRECONDITION(start >= end);
 
```

Comment:
This isn't a precondition, but a data_invariant.

Reply:
Fixed, will push once I've got a full update for this PR.